scripts/service.py

```diff
@@ -448,7 +450,25 @@ def format_token(t):
         else:
             self.token = None
 
+        self.session = session or requests.Session()
+
         self.namespace = namespace or global_config.namespace
+        
+    def _request(self, method, **kwargs):
+        \"\"\"Make a request using the session if enabled.\"\"\"
+        return self.session.request(method, **kwargs)
+    
+    def close(self):
+        \"\"\"Close the service session.\"\"\"
+        self.session.close()    
+    
+    def __enter__(self):
+        \"\"\"Open the service - session doesn't need to be opened.\"\"\"
+        return self
+        
+    def __exit__(self, *_):
+        \"\"\"Close the service.\"\"\"
+        self.session.close()
```

Comment:
In case we add any more teardown logic, I think we should use `close`

```suggestion
        self.close()
```